### DISCUSSION TITLE
Linting and Formatting

### DISCUSSION THREAD

**Discussion by rkaminsk** [potassco/python-project-template#11] [Ideas]
We currently use the following linters and formatters:

- flake8
- pylint
- mypy
- black
- autoflake

To reduce the number of config files, we can put there configuration into pyproject.toml.

There is also the ruff tool that combines much of the functionality of the above tools. It might be simpler to roll with one integrated tool. The performance advantage should not matter for our typical use cases.

**Comment by MaxOstrowski** [comment]
I usually like to go with the strictest option for the tools to not slack.
I ran ruff in parallel with pylint for a while now and it didn't give me any advantages. It is really fast, but it only checks a small subset of pylint (its more of a replacement to flake which I never used).
So I would vote for
 - pylint
 - mypy --strict
 - black
 - isort
just because these are the tools that I have experience with.

Pylint can get pretty slow and annoying at times but is still worth it for me. (30s to a minute to check my code)
